File: 0001-two-sum/0001-two-sum.java
class Solution {
        public int[] twoSum(int[] nums, int target) {
            HashMap<Integer, Integer> num_idx = new HashMap<>();
            for (int i = 0; i < nums.length; i++) {
                int search = target - nums[i];
                if (num_idx.containsKey(search) &&  num_idx.get(search) != i) {
                    return new int[]{i, num_idx.get(search)};
                }
                num_idx.put(nums[i], i);
            }
            return new int[]{};
        }
}

File: 0001-two-sum/0001-two-sum.py

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        """
        return indices of two nums such that they add up to target
        """
        dict = {}
        for i in range(len(nums)):
            if target-nums[i] in dict:
                return [i, dict[target-nums[i]]]
            dict[nums[i]] = i
        return []